pkg/querier/querier.go

```diff
@@ -79,14 +79,18 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 }
 
 // Validate the config
-func (cfg *Config) Validate() error {
+func (cfg *Config) Validate(httpServerWriteTimeout time.Duration) error {
 	// Ensure the config wont create a situation where no queriers are returned.
 	if cfg.QueryIngestersWithin != 0 && cfg.QueryStoreAfter != 0 {
 		if cfg.QueryStoreAfter >= cfg.QueryIngestersWithin {
 			return errBadLookbackConfigs
 		}
 	}
 
+	if cfg.EngineConfig.Timeout > httpServerWriteTimeout {
+		return fmt.Errorf("querier timeout (%s) must be lower than HTTP server write timeout (%s)", cfg.EngineConfig.Timeout, httpServerWriteTimeout)
```

Comment:
What do you think about moving this into the `pkg/mimir` package? Along with the [other validations](https://github.com/grafana/mimir/blob/cca6e117cc537e4b68f5d31e27ca7c7eeda43618/pkg/mimir/mimir.go#L207-L212) that rely on multiple configuration structs?

I think that way we'll have easier testing and arguably maybe better separation of concerns in the querier package.

Reply:
Godd sugesstion! Thanks!